packages/replay-internal/src/util/createPerformanceEntries.ts

```diff
@@ -175,38 +175,44 @@ function createResourceEntry(
  * Add a LCP event to the replay based on a LCP metric.
  */
 export function getLargestContentfulPaint(metric: Metric): ReplayPerformanceEntry<WebVitalData> {
-  return getWebVital(metric, 'largest-contentful-paint');
+  const lastEntry = metric.entries[metric.entries.length - 1] as (PerformanceEntry & { element?: Element }) | undefined;
+  const element = lastEntry ? lastEntry.element : undefined;
+  return getWebVital(metric, 'largest-contentful-paint', element);
 }
 
 /**
  * Add a CLS event to the replay based on a CLS metric.
  */
 export function getCumulativeLayoutShift(metric: Metric): ReplayPerformanceEntry<WebVitalData> {
-  return getWebVital(metric, 'cumulative-layout-shift');
+  return getWebVital(metric, 'cumulative-layout-shift', undefined);
 }
 
 /**
  * Add a FID event to the replay based on a FID metric.
  */
 export function getFirstInputDelay(metric: Metric): ReplayPerformanceEntry<WebVitalData> {
-  return getWebVital(metric, 'first-input-delay');
+  const lastEntry = metric.entries[metric.entries.length - 1] as (PerformanceEntry & { target?: Element }) | undefined;
+  const element = lastEntry ? lastEntry.target : undefined;
+  return getWebVital(metric, 'first-input-delay', element);
 }
 
 /**
  * Add an INP event to the replay based on an INP metric.
  */
 export function getInteractionToNextPaint(metric: Metric): ReplayPerformanceEntry<WebVitalData> {
-  return getWebVital(metric, 'interaction-to-next-paint');
+  const lastEntry = metric.entries[metric.entries.length - 1] as (PerformanceEntry & { target?: Element }) | undefined;
```

Comment:
Should we move this into a function?

Reply:
Only FID and INP have the same types, so I'm not sure we should. LCP has `element` instead of `target` and CLS has an array which contains `node`